### PR TITLE
Remove dependency to totp-lib

### PR DESCRIPTION
**Pull request by KarolJastrzebski**
Adapt open-source solution from `com.warrenstrange:googleauth` and use it
as a replacement for `totp-lib`.

Making changes as small as possible and letting tests confirm stability.

Copied interface from `totp-lib` to the package that was using it.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.auth.totp;
+
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import com.warrenstrange.googleauth.IGoogleAuthenticator;
+
+public class StrangeTotpAuthenticator implements TotpAuthenticator {
```
Comment (rkondratowicz) [inline token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java]: Perhaps `TotpAuthenticatorImpl`?

File: token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Override
+    public boolean isOneTimePasswordValid(String base32Key, String token) {
+        try {
+            return googleAuthenticator.authorize(base32Key, Integer.parseInt(token));
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+}
```
Comment (rkondratowicz) [inline token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java]: add tests

File: api/build.gradle [hunk 3]
```diff
@@ -30,7 +30,8 @@ sourceSets {
   }
 }
 dependencies {
-  compile group: 'uk.gov.hmcts.auth.totp', name: 'totp', version: '1.0.1'
+  compile project(':service-token-generator')
```
Comment (rkondratowicz) [inline api/build.gradle]: is this needed?

Comment (KarolJastrzebski) [reply api/build.gradle]: Absolutely not. Removed in 845b3ce.

Comment (KarolJastrzebski) [reply api/build.gradle]: At this moment -- yes. Will be removed in the upcoming changes.

File: token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java (hunk 1 above)
Comment (KarolJastrzebski) [reply token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java]: This class uses `GoogleAuthenticator` implementation from Warren Strange, and I couldn't resist naming it after him.

File: token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java (hunk 2 above)
Comment (KarolJastrzebski) [reply token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java]: Thanks, added! :+1:

File: token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java (hunk 1 above)
Comment (KarolJastrzebski) [reply token-generator/src/main/java/uk/gov/hmcts/auth/totp/StrangeTotpAuthenticator.java]: Ok, following team/org naming profile. Still, previous name was **WAY** better.

File: token-generator/src/main/java/uk/gov/hmcts/auth/totp/TotpAuthenticatorImpl.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    public String issueOneTimePassword(String base32Key) {
+        return String.valueOf(googleAuthenticator.getTotpPassword(base32Key));
+    }
+
+    @Override
+    public boolean isOneTimePasswordValid(String base32Key, String token) {
+        try {
+            return googleAuthenticator.authorize(base32Key, Integer.parseInt(token));
+        } catch (NumberFormatException e) {
+            return false;
```
Comment (rkondratowicz) [inline token-generator/src/main/java/uk/gov/hmcts/auth/totp/TotpAuthenticatorImpl.java]: worth logging it

Comment (KarolJastrzebski) [reply token-generator/src/main/java/uk/gov/hmcts/auth/totp/TotpAuthenticatorImpl.java]: Thanks, done.

**Pull request merged**